modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/data_set/filters/modals/selection_filter/SelectionFilter.tsx

```diff
@@ -281,113 +284,122 @@ function Body({
-		if (filter) {
-			const selectionFilter = filter as ISelectionFilter;
+		const selectionFilter = filter as ISelectionFilter;
 
-			if (
-				selectionFilter.sourceType ===
-				ESelectionFilterSourceType.API_HEADLESS
-			) {
-				setSelectedRESTApplication(selectionFilter.restApplication);
-				setSelectedRESTSchema(selectionFilter.restSchema);
-				setSelectedRESTEndpoint(selectionFilter.restEndpoint);
+		if (
+			selectionFilter.sourceType ===
+			ESelectionFilterSourceType.API_HEADLESS
+		) {
+			const selectionFilter = filter as ISelectionFilter;
 
-				setSelectedItemKey(selectionFilter.itemKey);
-				setSelectedItemLabel(selectionFilter.itemLabel);
+			setSelectedRESTApplication(selectionFilter.restApplication);
+			setSelectedRESTSchema(selectionFilter.restSchema);
+			setSelectedRESTEndpoint(selectionFilter.restEndpoint);
 
-				setSource(selectionFilter.source);
-			}
-		}
-	}, [filter]);
+			setSelectedItemKey(selectionFilter.itemKey);
+			setSelectedItemLabel(selectionFilter.itemLabel);
 
-	useEffect(() => {
-		if (filter) {
-			const selectionFilter = filter as ISelectionFilter;
+			setSource(selectionFilter.source);
 			setSourceType(selectionFilter.sourceType);
 
-			let validSavedPreselectedValues: any[] = [];
-
-			if (
-				source &&
-				sourceType === ESelectionFilterSourceType.API_HEADLESS
-			) {
+			getAPISourceItems({
+				preselectedValueInput: '',
+				selectedItemKey: selectionFilter.itemKey,
+				selectedItemLabel: selectionFilter.itemLabel,
+				source: selectionFilter.source,
+			}).then((sourceItems) => {
 				const filterPreselectedValues = JSON.parse(
-					(filter as ISelectionFilter).preselectedValues || '[]'
+					selectionFilter.preselectedValues || '[]'
 				);
+				let validSavedPreselectedValues: any[] = [];
 
-				validSavedPreselectedValues = filteredSourceItems.filter(
-					(item) =>
+				validSavedPreselectedValues = sourceItems.filter(
+					(item: {label: string; value: string}) =>
 						filterPreselectedValues.find(
 							(filterValue: {label: string; value: string}) =>
 								filterValue.value === item.value
 						)
 				);
-			}
 
-			if (source && sourceType === ESelectionFilterSourceType.PICKLIST) {
-				validSavedPreselectedValues = (
-					source as IPickList
-				).listTypeEntries.filter((item) =>
-					JSON.parse(
-						(filter as ISelectionFilter).preselectedValues || '[]'
-					).includes(item.externalReferenceCode)
+				setPreselectedValues(validSavedPreselectedValues);
+				setIncludeMode(
+					validSavedPreselectedValues?.length
+						? filter && selectionFilter.include
+							? 'include'
+							: 'exclude'
+						: 'include'
 				);
-			}
+			});
+		}
 
-			setPreselectedValues(validSavedPreselectedValues);
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, []);
 
-			setIncludeMode(
-				validSavedPreselectedValues?.length
-					? filter && (filter as ISelectionFilter).include
-						? 'include'
-						: 'exclude'
-					: 'include'
-			);
+	useEffect(() => {
```

Comment:
There seems to be a plague on unnecessary `useEffect`s in filters implementation. This is a bad pattern that leads to bugs, difficulty in debugging, poor readability and poor performance. We should work towards removing them all.